libs/application/template-api-modules/src/lib/modules/templates/car-recycling/car-recycling.service.ts

```diff
@@ -0,0 +1,129 @@
+import { TemplateApiError } from '@island.is/nest/problem'
+import { TemplateApiModuleActionProps } from '../../../types'
+import { BaseTemplateApiService } from '../../base-template-api.service'
+
+@Injectable()
+export class CarRecyclingService extends BaseTemplateApiService {
+  constructor(
+    @Inject(LOGGER_PROVIDER) private logger: Logger,
+    private carRecyclingService: CarRecyclingClientService,
+  ) {
+    super(ApplicationTypes.CAR_RECYCLING)
+  }
+
+  async createOwner({ application, auth }: TemplateApiModuleActionProps) {
+    const { applicantName } = getApplicationExternalData(
+      application.externalData,
+    )
+
+    const response = await this.carRecyclingService.createOwner(
+      auth,
+      applicantName,
+    )
+
+    if (response.errors) {
+      throw new TemplateApiError(
+        {
+          title: errorMessages.errorTitle,
+          summary: errorMessages.createOwnerDescription,
+        },
+        500,
+      )
+    }
+  }
+
+  async createVehicle(auth: User, vehicle: VehicleMiniDto) {
+    if (vehicle && vehicle.permno) {
+      return await this.carRecyclingService.createVehicle(auth, vehicle.permno)
+    }
+  }
+
+  async recycleVehicles(
+    auth: User,
+    vehicle: VehicleMiniDto,
+    recyclingRequestType: RecyclingRequestTypes,
+  ) {
+    if (vehicle && vehicle.permno) {
+      return await this.carRecyclingService.recycleVehicle(
+        auth,
+        vehicle.permno,
+        recyclingRequestType,
+      )
+    }
+  }
+
+  async sendApplication({ application, auth }: TemplateApiModuleActionProps) {
+    const { selectedVehicles, canceledVehicles } = getApplicationAnswers(
+      application.answers,
+    )
+
+    let isError = false
+    try {
+      selectedVehicles.forEach(async (vehicle) => {
+        if (!isError) {
+          // Create vehicle
+          const vechicleResponse = await this.createVehicle(auth, vehicle)
+
+          if (vechicleResponse && vechicleResponse.errors) {
+            isError = true
+            this.logger.error(vechicleResponse.errors)
+          }
+
+          if (!isError) {
+            // Recycle vehicle
+            const response = await this.recycleVehicles(
+              auth,
+              vehicle,
+              RecyclingRequestTypes.pendingRecycle,
+            )
+
+            if (response && response.errors) {
+              isError = true
+              this.logger.error(response.errors)
+            }
+          }
+        }
+      })
+    } catch (error) {
+      isError = true
+      this.logger.error(error.messages)
+    }
+    try {
+      canceledVehicles.forEach(async (vehicle) => {
+        // Cancel recycling
+        this.recycleVehicles(auth, vehicle, RecyclingRequestTypes.cancelled)
+      })
+    } catch (error) {
+      isError = true
+      this.logger.error(error.messages)
+    }
+
+    return new Promise((resolve, reject) => {
+      // Simulate an asynchronous operation (e.g., fetching data)
+      setTimeout(() => {
```

Comment:
Why the extra timeout ?

Reply:
Im trying to prevent the application to be sent if there is an error coming from the backend.  Since we are making more than one call to the backend when sending the application I don't have a promise to return. 
It seems that if I removed the setTimeout then the application is submitted and no error is shown.

Reply:
If waiting for all operations to complete is the goal, it should be tied to the actual completion of the operations, not an arbitrary timeout. Suggest just using a promise.all to properly await all executions and resolve accordingly.

Reply:
I have made changes to the code as you suggested and was able to remove the timeout